### ISSUE TITLE
PDF XSS in admin/upload.php

### ISSUE THREAD

**Issue by yangwancheng** [GetSimpleCMS/GetSimpleCMS#1320]
Version：3.4.0
Payload: app.alert('xss');
Insert xss malicious code in pdf with pdf editor
![xss4](https://user-images.githubusercontent.com/30540295/69219648-3b942100-0bae-11ea-8ad9-ccda8b374ff9.png)
Access the file upload function of GetSimpleCMS, upload a malicious pdf, and get the uploaded file path in the response package.
![xss5](https://user-images.githubusercontent.com/30540295/69219851-b52c0f00-0bae-11ea-8818-2fda21197448.png)
Using google browser to access the path of the malicious pdf file upload will trigger xss, which must be accessed by google browser.
![xss3](https://user-images.githubusercontent.com/30540295/69220039-0fc56b00-0baf-11ea-83d6-f4d5960e211f.png)

**Comment by tablatronix** [comment]
Is this a google chrome pdf reader exploit?

**Comment by bigin** [comment]
What's wrong with this? JS used in PDF as a part of form fill-out on web sites. You'll need to upload the PDF into the admin site, which would be a security issue.
You can change status to `WontFix` for it.

**Comment by tablatronix** [comment]
It seems to be a browser exploit right? How can we mitigate that?

**Comment by bigin** [comment]
There is no need for it. I don't think it's necessary to do anything, an alert message isn't XSS yet.

**Comment by jinnywc** [comment]
Specify the content type of the response, so that users can download and parse the PDF type file locally when browsing

**Comment by jinnywc** [comment]
Specify the content-type of the response, so that users can download and parse the PDF type file locally when browsing

**Comment by jinnywc** [comment]
Inserting the specified code can also obtain sensitive information such as cookies

**Comment by bigin** [comment]
Well, and how do you get the PDF on the server if you don't have administrator privileges?

**Comment by jinnywc** [comment]
After we create users, we upload malicious PDF files. The XSS can be triggered by recording the path of the PDF file and enticing the administrator or other users to access it

**Comment by tablatronix** [comment]
content type is a server config, not much we can do about it in GS

**Comment by bigin** [comment]
hehe really dumb

**Comment by jinnywc** [comment]
It can be implemented by code(example：https://www.helplib.com/Web_Development/article_4821)

**Comment by jinnywc** [comment]
lol？I also play lol  .game

**Comment by bigin** [comment]
You'll need to upload the PDF on the server, which would be a security issue. It's not an exploit.

**Comment by tablatronix** [comment]
Yeah but implementing a entire downloader file service and wrapping all downloads in code... Would probably add 3 more vectors , path traversal, injection etc.. meh.

We have white/black lists for uploads, and people can secure their server though the proper means for downloads.

**Comment by tablatronix** [comment]
I guess we could suggest or provide something for htaccess.. but not everyone runs apache

**Comment by bigin** [comment]
Yeah, custom upload restrictions, may be the best option at the moment. It's a single-admin cms (by default). And if a malicious user would get access to the admin, then we would probably have bigger problems than just XSS.

**Comment by tablatronix** [comment]
I will leave the discussion label on here for now, if anyone had ideas let me know. If not I will wontfix it
